zkevm-circuits/src/evm_circuit/param.rs

```diff
@@ -13,9 +16,11 @@ pub(crate) const MAX_GAS_SIZE_IN_BYTES: usize = 8;
 // Number of bytes that will be used of the address word.
 // If any of the other more signficant bytes are used it will
 // always result in an out-of-gas error.
-pub const NUM_ADDRESS_BYTES_USED: usize = 5;
-pub const MAX_MEMORY_SIZE_IN_BYTES: usize = 5;
-// Number of bytes that will be used of the JUMP* destination or code copy range
-// check. Although the deployed code has maximum size of 0x6000, the size of
-// a creation transaction could be 128KB, which needs 3 bytes to cover.
-pub const MAX_CODE_SIZE_IN_BYTES: usize = 3;
+pub(crate) const NUM_ADDRESS_BYTES_USED: usize = 5;
```

Comment:
irrelevant question:  does this address stands for wallet address ?  if so it is 160 bit length, 20 bytes

Reply:
it's memory address

Reply:
how to calculate 5 bytes of memory address ?

Reply:
It's explained [here](https://github.com/appliedzkp/zkevm-specs/blob/master/specs/opcode/51MLOAD_52MSTORE_53MSTORE8.md#circuit-behavior), but let me quote the main reason why it's 5 bytes:

> The memory expansion calculations and the memory addressing use a partial address value consisting of the 5 LSBs of address. If any other of the higher bytes are used an out-of-gas exception is always thrown.

Reply:
thanks I read it through  and think https://github.com/ethereum/go-ethereum/blob/master/core/vm/gas_table.go#L29#L40 constraint essentially , correct me if I am wrong .